spec/acceptance/class_spec.rb

```diff
@@ -1,16 +1,47 @@
 require 'spec_helper_acceptance'
 
-describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('osfamily')) do
-  it 'runs successfully' do
-    pp = <<-EOS
-    class { 'nodejs': }
-    if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
-      class { 'epel': }
-      Class['epel'] -> Class['nodejs']
-    }
-    EOS
+describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('os.family')) do
+  case fact('os.family')
+  when 'RedHat'
+    pkg_cmd = 'yum info nodejs | grep "^From repo"'
+  when 'Debian'
+    pkg_cmd = 'dpkg -s nodejs | grep ^Maintainer'
+  end
+
+  context 'default parameters' do
+    let(:pp) { "class { 'nodejs': }" }
+
+    it 'runs successfully' do
```

Comment:
I would add the shared example for this but I can submit a PR for that later.

Reply:
Cool, I didn't spend too long trying to get it to work (it wouldn't work inside an 'it' block).
I did see someone else create a `self.foo` in a test, and then use that where they wanted to reuse an item in the same test (vs. across different files).

Reply:
The trick is that it already contains it blocks so you don't need that.

Reply:
https://github.com/voxpupuli/puppet-nodejs/pull/313